lintipy.py

```diff
@@ -217,44 +219,56 @@ def run_process(self, code_path):
 
         """
         logger.info('Running: %s %s', self.cmd, ' '.join(self.cmd_args))
-        process = Popen(
-            ('python', '-m', self.cmd) + self.cmd_args,
-            stdout=PIPE, stderr=STDOUT,
-            cwd=code_path, env=self.get_env(),
-        )
-        process.wait()
-        info = resource.getrusage(resource.RUSAGE_CHILDREN)
-        log = process.stdout.read()
-        logger.debug(log)
-        logger.debug('exit %s', process.returncode)
-        logger.info('Saving log to S3')
-        key = os.path.join(self.label, self.full_name, "%s.log" % self.sha)
-        self.s3.put_object(
-            ACL='public-read',
-            Bucket=self.bucket,
-            Key=key,
-            Body=log,
-            ContentType='text/plain'
-        )
-        logger.info(
-            'linter exited with status code %s in %ss' % (process.returncode, info.ru_utime)
-        )
-        return (
-            info.ru_utime,
-            process.returncode,
-            "https://{0}.s3.amazonaws.com/{1}".format(self.bucket, key),
-        )
+        try:
+            process = subprocess.run(
+                ('python', '-m', self.cmd) + self.cmd_args,
+                stdout=subprocess.PIPE, stderr=subprocess.STDOUT,
+                cwd=code_path, env=self.get_env(),
+                timeout=self.cmd_timeout,
+            )
+        except subprocess.TimeoutExpired:
+            self.set_status(ERROR, 'Command timed out after %ss' % self.cmd_timeout,
+                            'https://lambdalint.github.io/#faq')
+            raise
+        else:
+            info = resource.getrusage(resource.RUSAGE_CHILDREN)
+            log = process.stdout
```

Comment:
You removed `.read()` here. Is it still working this way?

Reply:
It's not documented well, but [CompletedProcess.stdout](https://docs.python.org/3/library/subprocess.html#subprocess.CompletedProcess.stdout) is bytes not a stream.